hex/helpers/lib/run.exs

```diff
@@ -40,37 +41,115 @@ defmodule DependencyHelper do
     run_script("parse_deps.exs", dir)
   end
 
-  defp run(%{"function" => "get_latest_resolvable_version", "args" => [dir, dependency_name, credentials]}) do
-    run_script("check_update.exs", dir, [dependency_name] ++ credentials)
+  defp run(%{
+         "function" => "get_latest_resolvable_version",
+         "args" => [dir, dependency_name, credentials]
+       }) do
+    set_credentials(credentials)
+
+    run_script("check_update.exs", dir, [dependency_name])
   end
 
   defp run(%{"function" => "get_updated_lockfile", "args" => [dir, dependency_name, credentials]}) do
-    run_script("do_update.exs", dir, [dependency_name] ++ credentials)
+    set_credentials(credentials)
+
+    run_script("do_update.exs", dir, [dependency_name])
   end
 
   defp run_script(script, dir, args \\ []) do
-    args = [
-      "run",
-      "--no-deps-check",
-      "--no-start",
-      "--no-compile",
-      "--no-elixir-version-check",
-      script
-    ] ++ args
+    args =
+      [
+        "run",
+        "--no-deps-check",
+        "--no-start",
+        "--no-compile",
+        "--no-elixir-version-check",
+        script
+      ] ++ args
 
     System.cmd(
       "mix",
       args,
-      [
-        cd: dir,
-        env: %{
-          "MIX_EXS" => nil,
-          "MIX_LOCK" => nil,
-          "MIX_DEPS" => nil
-        }
-      ]
+      cd: dir,
+      env: %{
+        "MIX_EXS" => nil,
+        "MIX_LOCK" => nil,
+        "MIX_DEPS" => nil
+      }
     )
   end
+
+  defp set_credentials([]), do: :ok
+
+  defp set_credentials(["hex_organization", organization, token | tail]) do
+    url =
+      "hexpm"
+      |> Hex.Repo.get_repo()
+      |> Map.fetch!(:url)
+      |> URI.merge("/repos/#{organization}")
+      |> to_string()
+
+    update_repos("hexpm:#{organization}", %{url: url, public_key: nil, auth_key: token})
+
+    set_credentials(tail)
+  end
+
+  defp set_credentials(["hex_repo", repo, url, auth_key, fingerprint | tail]) do
+    case fetch_public_key(repo, url, auth_key, fingerprint) do
+      {:ok, public_key} ->
+        update_repos(repo, %{auth_key: auth_key, public_key: public_key, url: url})
+
+        set_credentials(tail)
+
+      error ->
+        handle_result(error)
+    end
+  end
+
+  defp set_credentials([_mode, org_or_url | _]) do
```

Comment:
This fall-through case catches when required options like the organization token are missing.